File: lib/models/planilla.dart
// lib/models/planilla.dart
import 'responsable.dart';

class Planilla {
  final int id;
  final String nombre;
  final DateTime creada;
  final List<Responsable> responsables;
  final String? asignadoA; // Usuario asignado (opcional, depende del backend)

  Planilla({
    required this.id,
    required this.nombre,
    required this.creada,
    required this.responsables,
    this.asignadoA,
  });

  factory Planilla.fromJson(Map<String, dynamic> json) => Planilla(
    id: json['id'] as int,
    nombre: json['nombre'] as String,
    creada: DateTime.parse(json['creada'] as String),
    responsables: (json['responsables'] as List?)
        ?.map((e) => Responsable.fromJson(e as Map<String, dynamic>))
        .toList() ?? [],
    asignadoA: (json['asignado_a'] as String?)
        ?? (json['asignadoA'] as String?)
        ?? (json['usuario'] as String?),
  );

  Map<String, dynamic> toJson() => {
    'id': id,
    'nombre': nombre,
    'creada': creada.toIso8601String(),
    'responsables': responsables.map((r) => r.toJson()).toList(),
    if (asignadoA != null) 'asignado_a': asignadoA,
  };
}

File: src/models/Planilla.ts
export interface Planilla {
    id: number;
    nombre: string;
    fecha: Date;
    responsableId: number;
    mascotas: number[];
}

export class PlanillaModel {
    static fromJson(json: any): Planilla {
        return {
            id: json.id || 0,
            nombre: json.nombre || '',
            fecha: json.fecha ? new Date(json.fecha) : new Date(),
            responsableId: json.responsable_id || 0,
            mascotas: json.mascotas || [],
        };
    }

    static toJson(planilla: Planilla): any {
        return {
            id: planilla.id,
            nombre: planilla.nombre,
            fecha: planilla.fecha.toISOString(),
            responsable_id: planilla.responsableId,
            mascotas: planilla.mascotas,
        };
    }
} 